src/Compilers/CSharp/Portable/Operations/CSharpOperationNodes.cs

```diff
@@ -1468,17 +1486,32 @@ internal sealed class CSharpLazyLocalFunctionOperation : LazyLocalFunctionOperat
     {
         private readonly CSharpOperationFactory _operationFactory;
         private readonly BoundLocalFunctionStatement _localFunctionStatement;
+        private readonly ImmutableArray<ConditionalOperation> _nullChecksToPrepend;
 
-        internal CSharpLazyLocalFunctionOperation(CSharpOperationFactory operationFactory, BoundLocalFunctionStatement localFunctionStatement, IMethodSymbol symbol, SemanticModel semanticModel, SyntaxNode syntax, ITypeSymbol type, Optional<object> constantValue, bool isImplicit) :
+        internal CSharpLazyLocalFunctionOperation(CSharpOperationFactory operationFactory, BoundLocalFunctionStatement localFunctionStatement, IMethodSymbol symbol, SemanticModel semanticModel, SyntaxNode syntax, ITypeSymbol type, ImmutableArray<ConditionalOperation> nullChecksToPrepend, Optional<object> constantValue, bool isImplicit) :
             base(symbol, semanticModel, syntax, type, constantValue, isImplicit)
         {
             _operationFactory = operationFactory;
             _localFunctionStatement = localFunctionStatement;
+            _nullChecksToPrepend = nullChecksToPrepend;
         }
 
         protected override IBlockOperation CreateBody()
         {
-            return (IBlockOperation)_operationFactory.Create(_localFunctionStatement.Body);
+            if (_nullChecksToPrepend.IsDefaultOrEmpty)
+            {
+                return (IBlockOperation)_operationFactory.Create(_localFunctionStatement.Body);
+            }
+
+            var constructed = _operationFactory.CreateFromArray<BoundStatement, IOperation>(_localFunctionStatement.Body.Statements).InsertRange(0, _nullChecksToPrepend);
+            return new BlockOperation(
+                constructed,
+                _localFunctionStatement.Body.Locals.CastArray<ILocalSymbol>(),
+                this.OwningSemanticModel,
+                this.Syntax,
+                this.Type,
+                this.ConstantValue,
+                isImplicit: true);
```

Comment:
Can this be extracted to a helper method used here and in `CSharpLazyAnonymousFunctionOperation`?